drivers/pds/api/dataServiceDeployment.go

```diff
@@ -62,13 +62,13 @@ func (ds *DataServiceDeployment) CreateDeployment(projectID string, deploymentTa
 }
 
 // CreateDeploymentWithScehduleBackup return newly created deployment model with schedule backup enabled.
-func (ds *DataServiceDeployment) CreateDeploymentWithScehduleBackup(projectID string, deploymentTargetID string, dnsZone string, name string, namespaceID string, appConfigID string, imageID string, nodeCount int32, serviceType string, resourceTemplateID string, storageTemplateID string, backupPolicyID string, backupTargetID string) (*pds.ModelsDeployment, error) {
+func (ds *DataServiceDeployment) CreateDeploymentWithSchehduleBackup(projectID string, deploymentTargetID string, dnsZone string, name string, namespaceID string, appConfigID string, imageID string, nodeCount int32, serviceType string, resourceTemplateID string, storageTemplateID string, backupPolicyID string, backupTargetID string) (*pds.ModelsDeployment, error) {
```

Comment:
can you please a PTX for optimising this function and CreateDeployment, make only one function CreateDeployment and take ControllersCreateProjectDeployment as input,
having these many parameters is not a good design

Reply:
Please find the ticket regarding optimization as follows: **https://portworx.atlassian.net/browse/PA-855**
Also, also since most of the changes are around the code improvement in existing code(i.e Error logging etc). Hence we will track the improvement as part of the following ticket **https://portworx.atlassian.net/browse/PA-678.**